### PR TITLE
chore(deps): update prisma to v5

### PR DESCRIPTION
**Pull request by overbit**
Close: #8117

## PR Details

Upgrade monorepo to prisma v5 to improve performance.
Comparison between same query on v4 and v5

V4 
Screenshot 2024-02-24 at 10 33 14

V5 
Screenshot 2024-02-24 at 10 33 39

## PR Checklist

- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Review by yuval-hazaz — COMMENTED** [review]
LGTM from a technical perspective.
I am not sure why the type changes are required as part of this PR (commit, resource.id, etc...) so I am assuming there is a reason.

I will leave it to @mulygottlieb to approve the PR based on his work plans and risk assessment

**Comment by mulygottlieb** [comment]
Thanks @overbit! Great initiative!
We will wait until the BTM release is out and stabilized and then incorporate this into our versions (probably early next week).

**Comment by overbit** [comment]
> LGTM from a technical perspective. I am not sure why the type changes are required as part of this PR (commit, resource.id, etc...) so I am assuming there is a reason.
> 
> I will leave it to @mulygottlieb to approve the PR based on his work plans and risk assessment

Prisma is more strict in v5 and it doesn't allow findUnique with potentially no inputs (that makes sense) and this drove the changes you noticed.

**Review by abrl91 — APPROVED** [review]

**Pull request merged**
